newt/project/project.go

```diff
@@ -499,16 +506,24 @@ func (proj *Project) verifyNewtCompat() error {
 
 // addRepo Adds an entry to the project's repo map.  It clones the repo if it
 // does not exist locally.
-func (proj *Project) addRepo(r *repo.Repo) error {
-	if err := r.EnsureExists(); err != nil {
-		return err
+func (proj *Project) addRepo(r *repo.Repo, download bool) error {
+	if download {
+		if err := r.EnsureExists(); err != nil {
+			return err
+		}
+	} else {
+		if !r.CheckExists() {
+			return util.NewNewtError(
+				fmt.Sprintf(
+					"Repo \"%s\" is not installed, please run `upgrade`!", r.Name()))
```

Comment:
Do you think this should say `newt upgrade`?  I guess the `newt` part is pretty obvious.  Up to you!

Reply:
I've found it too long, and I hoped nobody would actually type `$ upgrade` in the shell, but I changed it to show `newt upgrade`, it's probably better!